### PR TITLE
feat(scraper): Remove deprecated Google API job(s) and implement HTML scraping for google careers page

### LINKED ISSUE #99: bug: Google Careers /api/v3/search endpoint returns 404 — 0 Google jobs collected every run
## Problem

The Google Careers fetch phase silently returns **0 jobs on every run** because the scraped API endpoint is dead as of 2026-03.

```
❌ Google 'junior software engineer': 404 Client Error: Not Found
❌ Google 'new graduate software engineer': 404 Client Error: Not Found
...
✅ Google parallel fetch complete: 0 jobs from 8/8 searches
```

Both `fetch_google_jobs()` and `fetch_google_jobs_parallel()` attempt:
```
GET https://careers.google.com/api/v3/search/?location=United States&q=...&page_size=100
```

This returns HTTP 404 for all search terms.

## Short-term Fix (already landed)

Both functions now catch the 404 and emit an explicit error log instead of silently retrying:

```
❌ Google '{term}': careers API returned 404 — endpoint deprecated, see open GitHub issue
```

## Acceptance Criteria

- [ ] Identify the current working Google Careers job search endpoint (inspect network requests on careers.google.com/jobs/results)
- [ ] Update the URL and field mapping in both `fetch_google_jobs` and `fetch_google_jobs_parallel` (~lines 595 and 1015 in `update_jobs.py`)
- [ ] Add a smoke test asserting at least 1 Google job is returned for a known search term

**Difficulty:** Intermediate

### PR DESCRIPTION
**Pull request by half-measures**
Not scraping google did not sit right, so I made a PoC version to just scrape the google careers page. 
Google is not really new grad, everyone knows they have some very hard interviews but they do have a small amount of jobs on the Google Career page for 'new'. My own running of this showed of 31 jobs, 7 ended up in our final page.
 
- Uses the new URL direct from google.careers. 
- fragil, google is known to change and alter things which would break this function. I have tried to build in stops to stop garbage getting in or errors.
- Google has crazy fine detection, set a limit of 3 pages for now but it can be increased
- Currently returns 7 jobs to our list. We could increase it if we use internships or mid as mid for them is still without our 5 year cutoff.
- Has backoff logic for HTTP/URL errors.
- explicit handling of JSON decode errors 
- logging more explicit if anything goes wrong for easy tracking.

## Linked Issue

Fixes #99, May be its own issue?

## Summary

I removed the fetch_google_jobs_parallel and refactored fetch_google_jobs to just parse raw and deep nested html, and return it in the same format as the original function.
I enabled the config.yml flag for google to get it working once more.
I tried to remove mentions and old bits about the function. I did have to remove some tests.
Tests are very much needed for the new format but I wanted to make sure this worked first and get it out.

## Changes Made

| File | What changed |
|------|-------------|
| `scripts/update_jobs.py` |` Refactored fetch_google_jobs, removed fetch_google_jobs_parallel` |
| `config.yml` |`changed the google enabled flag to true` |
|`test_domain_limiter.py`| `removed some tests, removed mentions calling _parallel `|

## Testing

- [x] `python -m py_compile scripts/update_jobs.py` — no errors
- [x] `make test` — all tests pass
- [ ] `pre-commit run --all-files` — clean
Pre-commit seems to create 20k files in my commit for some reason. I do know the bots run this for me.

## Notes for Reviewer

URL; The URL is key and the OG is https://www.google.com/about/careers/applications/jobs/results?location=United%20States&target_level=EARLY

You can mess around easily with that, and the page only returns 20 results, but adding a PAGE=2 gets us page two of the same query. Currently with how easily google could block us, and the fact they may not whitelist github much on there networks, I have limited the (142 results for EARLY USA jobs as of March 16th 2026) to just 3 pages, so about 30/31. The func block goes through and adds to the page number depending on how many it finds but for now I think limiting is best while we find more kinks. 

I am aware this adds a single addition import, urlencode. Unsure if that is ok or I should find a work around to avoid doing so.
---

## Summary by CodeRabbit

* **New Features**
  * Google job listings enabled with multi-page pagination (configurable max pages, default 3)

* **Bug Fixes**
  * Cross-page and cross-term deduplication of job results
  * More robust parsing with fallbacks for incomplete or malformed entries
  * Improved retry/backoff and early abort on rate-limit responses

* **Tests**
  * Expanded suite covering parsing, pagination, deduplication, error handling, and edge cases

* **Chores**
  * Configuration and test updates to support the new Google source

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -77,3 +81,523 @@
...
+        patch('update_jobs.generate_health_json'),
+        patch('update_jobs.check_job_url_health'),
+        patch('builtins.open', new_callable=MagicMock), # Prevent writing files
+        patch('os.makedirs'),
+    ):
+        # We need to catch SystemExit if main fails due to missing files (README etc)
+        # But we only care about the coercion call
+        try:
+            main()
+        except SystemExit:
```
[the remark this answers is not in the document]
Comment (ambicuity) [reply tests/test_utils.py]: Patched in commit b583ff08: replaced the empty `except SystemExit: pass` with an explicit `except SystemExit as exc` assertion (`exc.code in (None, 0, 1)`) so test intent is preserved without a no-op handler.

**Review by ambicuity — APPROVED** [review]
Validated against issue #99 and latest head commit.\n\n- Local full suite passed (195 tests).\n- Live Google Careers smoke checks succeeded (AF_initDataCallback present; parser extracted jobs for representative terms).\n- Addressed remaining open review thread in tests by removing empty except handler.\n- Existing dual target_level behavior is intentional and currently returns broader relevant coverage for some terms.\n\nApproving for squash merge.
